server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3913,8 +3915,7 @@ private Pair<List<Long>, Integer> searchForServiceOfferingIdsAndCount(ListServic
             }
 
             serviceOfferingSearch.join("diskOfferingSearch", diskOfferingSearch, JoinBuilder.JoinType.INNER, JoinBuilder.JoinCondition.AND,
-                    serviceOfferingSearch.entity().getDiskOfferingId(), diskOfferingSearch.entity().getId(),
-                    serviceOfferingSearch.entity().setString("Active"), diskOfferingSearch.entity().getState());
+                    serviceOfferingSearch.entity().getDiskOfferingId(), diskOfferingSearch.entity().getId());
```

Comment:
Isn't this a query that is executed if vmid != null? if so, should we list inactive offerings?

Reply:
Since this PR aims to support disabling service offerings - to make it similar to the way network offerings work, it would make sense to list active and inactive offerings

Reply:
This method is too long (500+ lines...) for me to judge what it is exactly doing when. 
I just want to make sure that when creating VMs we don't list inactive offerings. If this change does not cause that, then I'm ok.

Reply:
Al, that's a fair point.. Probably reverting to it listing Active by default would be a better option. Thanks.